packages/vuetify/src/components/VFileInput/VFileInput.ts

```diff
@@ -0,0 +1,213 @@
+    },
+    isLabelActive (): boolean {
+      return this.isDirty
+    },
+    text (): string[] {
+      if (!this.isDirty) return [this.placeholder]
+
+      return this.internalValue.map((file: File) => {
+        const name = this.truncateText(file.name)
+
+        return !this.displaySize ? name : `${name} (${humanReadableFileSize(file.size, this.base)})`
+      })
+    },
+    base (): 1000 | 1024 | undefined {
+      return typeof this.displaySize !== 'boolean' ? this.displaySize : undefined
+    },
+    hasChips (): boolean {
+      return this.chips || this.smallChips
+    },
+  },
+
+  methods: {
+    clearableCallback () {
+      this.internalValue = []
+    },
+    genChips () {
+      if (!this.isDirty) return []
+
+      return this.text.map((text, i) => this.$createElement(VChip, {
+        props: { small: this.smallChips },
+        on: {
+          'click:close': () => {
+            this.internalValue.splice(i, 1)
+            this.internalValue = this.internalValue // Trigger the watcher
+          },
+        },
+      }, [text]))
+    },
+    genInput () {
+      const input = VTextField.options.methods.genInput.call(this)
+
+      input.data!.attrs!.multiple = this.multiple
+      input.data!.attrs!.accept = this.accept
+      delete input.data!.domProps!.value
+
+      return [this.genSelections(), input]
+    },
+    genPrependSlot () {
+      const icon = this.genIcon('prepend', () => {
+        this.$refs.input.click()
+      })
+
+      icon.data!.attrs! = { tabindex: 0 }
+
+      return this.genSlot('prepend', 'outer', [icon])
+    },
+    genSelectionText (): string[] {
+      const length = this.text.length
+
+      if (length < 2) return this.text
+      if (this.displaySize && !this.counter) return [this.counterValue]
+      return [this.$vuetify.lang.t(this.counterString, length)]
+    },
+    genSelections () {
+      const children = []
+
+      if (this.isDirty && this.$scopedSlots.selection) {
+        this.internalValue.forEach((file: File, index: number) => {
+          if (!this.$scopedSlots.selection) return
+
+          children.push(
+            this.$scopedSlots.selection({
+              text: this.text[index],
+              file,
+              index,
+            })
+          )
+        })
+      } else {
+        children.push(this.hasChips && this.isDirty ? this.genChips() : this.genSelectionText())
+      }
+
+      return this.$createElement('div', {
+        staticClass: 'v-file-input__text',
+        class: {
+          'v-file-input__text--placeholder': this.placeholder && !this.isDirty,
+          'v-file-input__text--chips': this.hasChips && !this.$scopedSlots.selection,
+        },
+        on: {
+          click: () => this.$refs.input.click(),
+        },
+      }, children)
+    },
+    onInput (e: Event) {
+      this.internalValue = [...(e.target as HTMLInputElement).files]
+    },
+    truncateText (str: string) {
+      if (str.length < 22) return str
```

Comment:
magical number?

Reply:
The number of characters before a normal file input truncates the text.

Reply:
but it's still magical number, even if the magic is explained
also it looks like truncated text is passed to the slot scope, it should get the full text so that dev could truncate it in his own way or not truncate at all

Reply:
I can prop the number. The scoped slot also passes file, which has the name property, which I felt was sufficient if the user wanted the non-truncated version.